presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -136,6 +137,25 @@ public void testJoinDynamicFilteringSingleValue()
         assertEquals(rowsRead, ImmutableSet.of(6L, buildSideRowsCount));
     }
 
+    @Test
+    public void testJoinDynamicFilteringMultiJoin()
```

Comment:
Since it's a planner bug, it will be better to add the test in `TestDynamicFilter` instead and verify that we get expected plan there. Once DynamicFiltersChecker is extended to cover this case, that will also ensure that we don't get beyond planning stage if wrong plan is generated.

Reply:
Sounds good - added `TestDynamicFilter#testNonPushedDownJoinFilterRemoval`.